[BUG] Can't resolve 'styles/annotations.css' when importing BokehJS

#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
* bokeh : 2.1.0
* node : 14.4.0
* react : 16.13.1
* typescript : 3.9.5
* OS : Ubuntu 18.04 in WSL2

#### Description of expected behavior and the observed behavior
An error occurred when just importing bokehJS in **webpack** based JSX code (React). It might be revisit of #8197 or webpack confused about `.css.js` (`style/annotation.css.js` )

Check this:
https://github.com/bokeh/bokeh/blob/00cdec4979bc27a4f1cb35dc3d8f4e62ae93deac/bokehjs/src/lib/models/annotations/annotation.ts#L9

#### Complete, minimal, self-contained example code that reproduces the issue

[Simple Repo to reproduce](https://github.com/appleparan/bokeh-8197)

What I did;
```
$ npx create-react-app myapp
$ cd myapp
$ npm i @bokeh/bokehjs
```
then add `import * as Bokeh from "@bokeh/bokehjs";` in `App.js`.

#### Stack traceback and/or browser JavaScript console output

```
./node_modules/@bokeh/bokehjs/build/js/lib/models/annotations/annotation.js
Module not found: Can't resolve 'styles/annotations.css' in 'mydirectories/node_modules/@bokeh/bokehjs/build/js/lib/models/annotations'
```

--- a/bokehjs/src/compiler/compiler.ts
+++ b/bokehjs/src/compiler/compiler.ts
@@ -105,7 +105,8 @@ export function default_transformers(options: ts.CompilerOptions): ts.CustomTran
         const module_file = join(base, module_path)
         if (ts.sys.fileExists(module_file) ||
             ts.sys.fileExists(module_file + ".ts") ||
-            ts.sys.fileExists(join(module_file, "index.ts"))) {
+            ts.sys.fileExists(join(module_file, "index.ts")) ||
+            options.outDir != null && ts.sys.fileExists(join(options.outDir, module_path + ".js"))) {
           const rel_path = normalize(relative(dirname(file), module_file))
           return rel_path.startsWith(".") ? rel_path : `./${rel_path}`
         }

